src/test/kotlin/org/malachite/estella/application/ApplicationsIntegration.kt

```diff
@@ -0,0 +1,184 @@
+import org.junit.jupiter.api.Order
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestMethodOrder
+import org.malachite.estella.BaseIntegration
+import org.malachite.estella.commons.EStellaHeaders
+import org.malachite.estella.commons.Message
+import org.malachite.estella.commons.UnauthenticatedMessage
+import org.malachite.estella.offer.infrastructure.HibernateOfferRepository
+import org.malachite.estella.people.domain.JobSeekerFilePayload
+import org.malachite.estella.people.domain.toJobSeekerDTO
+import org.malachite.estella.people.infrastrucutre.HibernateJobSeekerRepository
+import org.malachite.estella.people.infrastrucutre.HibernateUserRepository
+import org.malachite.estella.services.ApplicationService
+import org.malachite.estella.services.JobSeekerService
+import org.malachite.estella.services.OfferService
+import org.malachite.estella.util.EmailServiceStub
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.HttpMethod
+import org.springframework.http.HttpStatus
+import strikt.api.expect
+import strikt.api.expectThat
+import strikt.assertions.isEqualTo
+import strikt.assertions.isGreaterThan
+import strikt.assertions.isGreaterThanOrEqualTo
+import strikt.assertions.isNotNull
+
+@TestMethodOrder(MethodOrderer.OrderAnnotation::class)
+class ApplicationsIntegration : BaseIntegration() {
+
+    @Autowired
+    private lateinit var jobSeekerRepository: HibernateJobSeekerRepository
+
+    @Autowired
+    private lateinit var offerRepository: HibernateOfferRepository
+
+
+    @Test
+    @Order(1)
+    fun `should be able to apply for offer as logged in user`() {
+        EmailServiceStub.stubForSendEmail()
+
+        val offer = getOffer()
+        val jobSeeker = getJobSeeker()
+
+
+        val response = httpRequest(
+            path = "/api/applications/apply/${offer.id}/user",
+            method = HttpMethod.POST,
+            headers = mapOf(EStellaHeaders.jwtToken to getAuthToken(jobSeeker.user.mail, "a")),
+            body = mapOf("files" to setOf<JobSeekerFilePayload>())
+        )
+        expectThat(response.statusCode).isEqualTo(HttpStatus.CREATED)
+        val responseBody = (response.body as Map<String, Any>).toApplicationDTO()
+        val applications = getApplications()
+        applications.find { it.id == responseBody.id }.let {
+            expectThat(it).isNotNull()
+            it!!
+            expect {
+                that(it.applicationDate).isEqualTo(responseBody.applicationDate)
+                that(it.seekerFiles).isEqualTo(emptySet())
+                that(it.stage).isEqualTo(responseBody.stage)
+                that(it.status).isEqualTo(responseBody.status)
+                that(it.status).isEqualTo(responseBody.status)
+                that(it.jobSeeker).isEqualTo(jobSeeker.toJobSeekerDTO())
+            }
+        }
+    }
+
+    @Test
+    @Order(2)
+    fun `should not be able to apply for offer as not logged in user`() {
```

Comment:
weź proszę zmień nazwę tego testu, bo już Cię opierdzielić miałem, że 1/4 systemu miała się opierać na niezalogowanych użytkownikach, ale w porę endpoint sprawdziłem XD

Reply:
Yyy to jakiej nazwy oczekujesz ?